src/test/java/de/lh/tool/service/entity/impl/UserServiceTest.java

```diff
@@ -0,0 +1,257 @@
+package de.lh.tool.service.entity.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.Calendar;
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import de.lh.tool.domain.exception.DefaultException;
+import de.lh.tool.domain.exception.ExceptionEnum;
+import de.lh.tool.domain.model.PasswordChangeToken;
+import de.lh.tool.domain.model.User;
+import de.lh.tool.domain.model.User.Gender;
+import de.lh.tool.repository.UserRepository;
+import de.lh.tool.service.entity.interfaces.PasswordChangeTokenService;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceTest {
+	@Mock
+	private UserRepository userRepository;
+
+	@Mock
+	private PasswordChangeTokenService passwordChangeTokenService;
+
+	@Mock
+	private AuthenticationManager authenticationManager;
+
+	@Mock
+	private PasswordEncoder passwordEncoder;
+
+	@InjectMocks
+	private UserServiceImpl userService;
+
+	@BeforeEach
+	public void before() {
+		MockitoAnnotations.initMocks(userService);
+	}
+
+	@Test
+	public void testCreateUserNoEmail() {
+		try {
+			userService.createUser(new User());
+			fail("exception expected");
+		} catch (DefaultException e) {
+			assertEquals(ExceptionEnum.EX_USER_NO_EMAIL, e.getException());
+		}
+	}
```

Comment:
Folgendermaßen fände ichs ein wenig eleganter:

```suggestion
DefaultException thrown = assertThrows(DefaultException.class, () -> {
	userService.createUser(new User());
});
assertEquals(ExceptionEnum.EX_USER_NO_EMAIL, thrown.getException());
```

Reply:
nice, assertThrows kannt ich gar nicht, man lernt nie aus ;)